### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by zfreiter**
01-flexbox.md - Text solutions added.
02-cssgrid.md - Text solutions added.

 03-count -
- Bootstrap added to html.
- Created a function to highlight all character's that a user enters.

04-background - 
- Bootstrap added to html.
- Created a function to generate a random color.
- Created a function that takes a positive value and uses that value to determine when the background color will change.

05-form - 
- Bootstrap added to html.
- Added an submit event listener for the form. It will display what the user inputs.

### CODE REVIEW

File: hw2/04-background.js [hunk 1]
```diff
@@ -1 +1,29 @@
...
+const body = document.querySelector('body');
+button.addEventListener('click', changeState);
+
+/* Sets a random background color */
+function colorChange() {
+  body.style.backgroundColor = `rgba(${Math.floor(
+    Math.random() * 255
+  )}, ${Math.floor(Math.random() * 255)}, ${Math.floor(
+    Math.random() * 255
+  )}, ${Math.floor(Math.random() * 100)}%)`;
```
Comment (Iwasson) [inline hw2/04-background.js:13]: By default Math.random() returns a random float between 0 and 1. By default rgba uses a decimal number for opacity so you can simplify this code quite a bit by removing the percentage and flooring

File: hw2/04-background.js [hunk 2]
```diff
@@ -1 +1,29 @@
...
+function changeState() {
+  if (input.value > 0) {
+    if (button.classList.contains('btn-primary')) {
+      button.classList = 'btn btn-danger mt-4';
+      button.innerHTML = 'Stop';
+      /* CALL setInterval to run colorChange function */
+      setInterval(colorChange, input.value * 1000);
+    } else {
+      button.classList = 'btn btn-primary mt-4';
+      /* Reloads the page to cancel the setInterval */
```
Comment (Iwasson) [inline hw2/04-background.js:25]: You should use clearInterval(colorChange) here instead of reloading the page.

File: hw2/05-form.js [hunk 3]
```diff
@@ -1 +1,41 @@
...
+    programmingLanguages ? 'Yes' : 'No',
+    '\nOperating Systems Course: ',
+    operatingSystems ? 'Yes' : 'No',
+    '\nFull Stack Course: ',
+    fullStack ? 'Yes' : 'No',
+    '\nAnything else: ',
+    textBox ? textBox : 'Nothing entered'
+  );
+
+  /* RESET FORM VALUES */
```
Comment (Iwasson) [inline hw2/05-form.js:33]: you can simplify this down to document.getElementById("myform").reset();

File: hw2/04-background.js (hunk 1 above)
Comment (zfreiter) [reply hw2/04-background.js:13]: Good catch.

File: hw2/04-background.js (hunk 2 above)
Comment (zfreiter) [reply hw2/04-background.js:25]: I used window.location.reload(true); because it was a listed resource.

File: hw2/05-form.js (hunk 3 above)
Comment (zfreiter) [reply hw2/05-form.js:33]: Good suggestions.

File: hw2/05-form.html [hunk 4]
```diff
@@ -7,13 +7,112 @@
...
+        <div class="mb-3">
+          <label for="email" class="fw-bold form-label">Email:</label>
+          <input
+            type="email"
+            class="form-control"
+            id="email"
+            aria-describedby="emailHelp"
+            required
+          />
+        </div>
```
Comment (ssiddhi) [inline hw2/05-form.html:42]: Your code is giving accessibility error: Broken ARIA reference. You might want to fix it by giving element referenced in the aria-describedby, a proper label. Everything else looks good. Code is responsive, works on different browsers and looks similar to the screenshot in pdf. Great job 👍

File: hw2/05-form.js [hunk 5]
```diff
@@ -1 +1,34 @@
...
+    '\nOperating Systems Course: ',
+    operatingSystems ? 'Yes' : 'No',
+    '\nFull Stack Course: ',
+    fullStack ? 'Yes' : 'No',
+    '\nAnything else: ',
+    textBox ? textBox : 'Nothing entered'
+  );
+
+  appForm.reset();
+});
```
Comment (ssiddhi) [inline hw2/05-form.js:34]: Code is working as expected. Nice work on this exercise 👍

**Review by ssiddhi — COMMENTED** [review]
Hey Zachary,
I have provided my review on this pull request.

File: hw2/03-count.html [hunk 6]
```diff
@@ -6,16 +6,20 @@
...
+      <h1 class="align-self-center fw-bold">Exercise 03 - Count</h1>
       <hr />
       <div>
-        <label>Highlight all occurences of a word:</label>
-        <input />
+        <label for="wordInput">Highlight all occurences of a word:</label>
+        <input id="wordInput" />
       </div>
       <hr />
       <div>
```
Comment (rutujapadgilwar) [inline hw2/03-count.html:25]: Everything looks fine, no changes.

File: hw2/03-count.js
Comment (rutujapadgilwar) [inline hw2/03-count.js:1]: Working perfectly now. You did a good job by adding the required changes. Nice work with exercise 03.

File: hw2/04-background.html [hunk 7]
```diff
@@ -6,23 +6,27 @@
...
     <link
       rel="stylesheet"
-      href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" />
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css"
+    />
     <script defer src="04-background.js"></script>
   </head>
-  <body>
-    <main>
+  <body style="background-color: #6c61db">
```
Comment (rutujapadgilwar) [inline hw2/04-background.html:13]: You did a perfect job adding the exact background color as the screenshot shows.

File: hw2/04-background.html [hunk 8]
```diff
@@ -6,23 +6,27 @@
...
+    <main
+      style="max-width: 520px"
+      class="d-flex flex-column justify-content-center rounded m-auto mt-5 p-3 mr-1 bg-white"
```
Comment (rutujapadgilwar) [inline hw2/04-background.html:16]: In the screenshot, the background color is "bg-light"

File: hw2/04-background.js
Comment (rutujapadgilwar) [inline hw2/04-background.js:1]: The code looks good!

File: hw2/05-form.html
Comment (rutujapadgilwar) [inline hw2/05-form.html:1]: Now everything looks good to me.

**Review by rutujapadgilwar — CHANGES_REQUESTED** [review]
Exercise 01, 02 --> perfectly fine.
Exercise 03, 04, 05 --> I saw you made changes and now everything looks good to me. Just one background color suggestion in exercise 04.

**Pull request merged**